flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -285,7 +281,13 @@ public LogicalSlot getAssignedResource() {
 
 	public Optional<ResultPartitionDeploymentDescriptor> getResultPartitionDeploymentDescriptor(
 			IntermediateResultPartitionID id) {
-		return Optional.ofNullable(producedPartitions.get(id));
+		Collection<ResultPartitionDeploymentDescriptor> trackedDeploymentDescriptors = vertex.getExecutionGraph().getPartitionTracker().getTrackedDeploymentDescriptors(getAttemptId());
+		for (ResultPartitionDeploymentDescriptor resultPartitionDeploymentDescriptor : trackedDeploymentDescriptors) {
```

Comment:
I hate this bit, but really don't want to introduce an additional map _just_ to support this lookup. Any ideas?

Reply:
anyways I think the method rather belongs to the tracker with the exec id added to the signature